src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -486,7 +486,7 @@ private bool TryInsert(TKey key, TValue value, InsertionBehavior behavior)
             Debug.Assert(entries != null, "expected entries to be non-null");
 
             IEqualityComparer<TKey>? comparer = _comparer;
-            uint hashCode = (uint)((comparer == null) ? key.GetHashCode() : comparer.GetHashCode(key));
+            uint hashCode = (uint)(comparer?.GetHashCode(key) ?? key.GetHashCode());
```

Comment:
I don't think this is an improvement.  The previous code was easy to understand: if comparer is null, do one thing, otherwise do something else.  This takes longer to visually parse and understand.  Please revert this.